release/models/platform/openconfig-platform.yang

```diff
@@ -403,12 +409,37 @@ module openconfig-platform {
 
     leaf location {
       type string;
+      status deprecated;
       description
         "System-supplied description of the location of the
         component within the system. This could be a bay position,
         slot number, socket location, etc. For component types that
         have an explicit slot-id attribute, such as linecards, the
-        system should populate the more specific slot-id.";
+        system should populate the more specific slot-id.
+
+        This leaf is deprecated and replaced by parent-position.";
+    }
+
+    leaf parent-position {
+      type string;
+      description
+        "System-supplied index to a position, port, slot, bay,
+        socket, etc.  This string must only indicate the name
+        of the position, and not any indication of the name of the
+        parent component within the system.  Instead, parent
+        component name should be present in the 'parent' leaf.
+
+        Typically the parent-position is a number, but it is observed
+        that some devices may use letters or alphanumerics.  The
+        position name should be the same name used to physically
+        identify the position in documentation or printed on the
+        device.
+
+        Some components with a parent may not have a parent-position.
+
+        For component types that have an explicit slot-id attribute,
+        such as linecards, the system should populate the more
+        specific slot-id.";
```

Comment:
The part about "the more specific slot-id" is not entirely clear to me.

If I have a flat linecard numbering schema, for example, `linecard1` to `linecard9`, both slot-id and parent-position will indicate a single number [1,9].

Reply:
This part of the description (carried over from the `location` leaf) is advising that component type `LINECARD` should use [/components/component/linecard/state/slot-id](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-platform.html#components-component-linecard-state-slot-id) instead of parent-position.   

However now that you point it out, I see parent-position and slot-id are redundant.  We could expand the scope of this change to deprecate slot-id as well since it has the same meaning.  WDYT?

Reply:
It seems reasonable for simple use cases like in the example above.

One scenario that we need to think about is when more complex numbering schemas are in use: some platforms have parent linecards with sub-slots (that contain Ethernet interfaces).  
In that case, something like `linecardX/Y` is typically used to identify the subslots, and there may be a divergence (e.g. a subslot can be modeled as a linecard with slot-id `x/y`, but the parent location will be `y`). Perhaps this text was added to describe such cases, and can be rephrased to better reflect that.

Reply:
I think the added `install-component` leaf will provide a clear and direct mapping to the component where the insertion/install occurs.  The implementation can choose to model the "subslot" as a component whose parent is the LINECARD and has children of type PORT.